lib/components/SelectionControl/toggle-switch/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from "react";
+import "./toggle-switch.scss";
+import clsx from "clsx";
+
+export interface SwitchToggleProps {
+    disabled?: boolean;
+    defaultChecked?: boolean;
+    onChange?: (isEnabled: boolean) => void;
+    className?: string;
+    id?: string;
+    onFocus?: () => void;
+    onBlur?: () => void;
+    checked?: boolean;
+}
+
+const ToggleSwitch = ({
+    disabled = false,
+    onChange,
+    className,
+    defaultChecked,
+    onFocus,
+    onBlur,
+    id,
+    checked,
+}: SwitchToggleProps) => {
+    const [isEnabled, setIsEnabled] = useState(defaultChecked || false);
+
+    useEffect(() => {
+        if (checked !== undefined) {
+            setIsEnabled(checked);
+        }
+    }, [checked]);
+
+    const toggleSwitch = () => {
+        if (!disabled) {
+            if (checked === undefined) {
+                const toggleEnable = !isEnabled;
+                setIsEnabled(toggleEnable);
+                onChange?.(toggleEnable);
+            } else {
+                onChange?.(!checked);
+            }
+        }
+    };
+
+    const handleKeyDown = (event: React.KeyboardEvent) => {
+        if (event.key === "Enter" || event.key === " ") {
+            event.preventDefault();
+            toggleSwitch();
+        }
+    };
+
+    return (
+        <button
+            className={clsx("toggle-switch", className)}
+            onClick={toggleSwitch}
+            data-testid='toggle-switch'
+            onKeyDown={handleKeyDown}
+            onFocus={onFocus}
+            id={id}
+            onBlur={onBlur}
+            disabled={disabled}
+            aria-pressed={isEnabled}
+        >
+            <div
+                className={clsx("toggle-switch__slider", {
+                    enabled: isEnabled,
```

Comment:
Another stupid question from me: `isEnabled` is indication if it's checked, correct? Not the opposite for `disabled`

Reply:
Yupp that's correct